src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -116,18 +186,24 @@ int main(int argc, char* argv[])
     // clean-up (close files and free memory)
     CAudioFileIf::destroy(phAudioFile);
     hOutputFile.close();
+    phOutputAudioFile->closeFile();
+    CAudioFileIf::destroy(phOutputAudioFile);
 
     for (int i = 0; i < stFileSpec.iNumChannels; i++)
         delete[] ppfAudioData[i];
     delete[] ppfAudioData;
     ppfAudioData = 0;
 
+    for (int i = 0; i < stFileSpec.iNumChannels; i++)
+        delete[] ppfOutputData[i];
+    delete[] ppfOutputData;
+    ppfOutputData = 0;
+
     // all done
     return 0;
```

Comment:
Not sure how you verify the results. Maybe restore or print out the status so that user can have a quick view. If necessary, user can go on to check the recorded info or audio.